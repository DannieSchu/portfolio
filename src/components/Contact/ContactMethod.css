.ContactMethod {
  display: flex;
  text-decoration: none;
  z-index: 1;
  height: 3.5rem;

  img {
    width: 2.5rem;
    height: 2.5rem;
    margin-right: 2rem;
    transition: all .2s ease-in-out;
    
    &:hover {
      transform: scale(1.1);
    }
  }

  p {
    font-size: 1.6rem;
    position: relative;
    bottom: 1rem;

    &:hover {
      color: var(--gold)
    }
  }

  @media only screen and (max-width: 768px) {
    height: 1.8rem;

    img {
      width: 1.5rem;
      height: 1.5rem;
      margin-right: 1rem;
    }

    p {
      font-size: 1rem;
    }
  }
}
