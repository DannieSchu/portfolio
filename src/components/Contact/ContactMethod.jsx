import React from 'react';
import PropTypes from 'prop-types';
import styles from './ContactMethod.css';
import typeStyles from '../../styles/type.css';

const ContactMethod = ({ image, link, text }) => {
  return (
    <a href={link} className={styles.ContactMethod}>
      <img src={image} alt={text} />
      <p className={typeStyles.body}>{text}</p>
    </a>
  );
};

ContactMethod.propTypes = {
  image: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default ContactMethod;
