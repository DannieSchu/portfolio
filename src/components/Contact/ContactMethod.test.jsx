import React from 'react';
import { shallow } from 'enzyme';
import ContactMethod from './ContactMethod';

describe('ContactMethod component', () => {
  it('renders ContactMethod', () => {
    const wrapper = shallow(<ContactMethod image="../assets/social-media/twitter.png" link="https://twitter.com/DannieSchumaker" text="dannieschumaker" />);
    expect(wrapper).toMatchSnapshot();
  });
});
