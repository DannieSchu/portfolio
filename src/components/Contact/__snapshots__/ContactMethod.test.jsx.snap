// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMethod component renders ContactMethod 1`] = `
<a
  className="ContactMethod"
  href="https://twitter.com/DannieSchumaker"
>
  <img
    alt="dannieschumaker"
    src="../assets/social-media/twitter.png"
  />
  <p
    className="body"
  >
    dannieschumaker
  </p>
</a>
`;
